app/clients/document_download.py

```diff
@@ -23,17 +25,22 @@ def init_app(self, app):
     def get_upload_url(self, service_id):
         return "{}/services/{}/documents".format(self.api_host, service_id)
 
-    def upload_document(self, service_id, file_contents, is_csv=None):
+    def upload_document(self, service_id, file_contents, is_csv=None, verification_email: Optional[str] = None):
```

Comment:
what does `Optional` give us compared to if we did not use it? I haven't seen it before.
Is it to make Python more statically typed? If so, do you think it's a direction we should take overall, and if so, why? Thanks!

Reply:
For now, it just tells us (devs) that the value can be a string or it can be None and to handle that appropriately. I'd like for us to move towards having type hints in the code base and having these statically checked with mypy, but something to discuss more broadly before committing to it.

Reply:
Ok, interesting, thanks! Yeah, definitely one to discuss on a tech catch up when David and Leo are back - or maybe even we could have a tech away day to discuss tech strategy in an ambitious way?

Reply:
That's a great idea, we should totally do that 👍